shared/chat/conversation/info-panel/add-to-channel/index.js

```diff
@@ -0,0 +1,158 @@
+// @flow
+import * as React from 'react'
+import * as I from 'immutable'
+import * as Kb from '../../../../common-adapters'
+import * as Styles from '../../../../styles'
+import {smartPluralize} from '../../../../util/string'
+
+type Props = {|
+  error: ?string,
+  onCancel: () => void,
+  onSubmit: (usernames: Array<string>) => void,
+  title: string,
+  users: Array<{alreadyAdded: boolean, fullname: string, username: string}>,
+  waitingKey: string,
+|}
+
+type State = {|
+  selected: I.Set<string>,
+|}
+
+class AddToChannel extends React.Component<Props, State> {
+  state = {selected: I.Set()}
+  _itemHeight = {
+    height: Styles.isMobile ? 48 : 40,
+    type: 'fixed',
+  }
+
+  _toggleSelected = username =>
+    this.setState(s => ({
+      selected: s.selected.includes(username) ? s.selected.remove(username) : s.selected.add(username),
+    }))
+
+  _renderItem = (idx, user) => (
+    <Kb.ListItem2
+      {...(user.alreadyAdded ? {} : {onClick: () => this._toggleSelected(user.username)})}
+      firstItem={idx === 0}
+      type="Small"
+      icon={<Kb.Avatar username={user.username} size={32} />}
+      body={
+        <Kb.Box2 direction="horizontal" fullWidth={true} fullHeight={true} alignItems="center">
+          <Kb.Box2 direction="vertical" fullWidth={true}>
+            <Kb.ConnectedUsernames
+              colorFollowing={true}
+              type="BodySemibold"
+              usernames={[user.username]}
+              underline={false}
+            />
+            {!user.alreadyAdded && !!user.fullname && (
+              <Kb.Text lineClamp={1} type="BodySmall" style={styles.userSubtext}>
+                {user.fullname}
+              </Kb.Text>
+            )}
+            {user.alreadyAdded && (
+              <Kb.Text lineClamp={1} ellipsizeMode="head" type="BodySmall" style={styles.userSubtext}>
+                {user.username} is already a member
+              </Kb.Text>
+            )}
+          </Kb.Box2>
+        </Kb.Box2>
+      }
+      action={
+        <Kb.Checkbox
+          disabled={user.alreadyAdded}
+          checked={user.alreadyAdded || user.selected}
+          onCheck={() => this._toggleSelected(user.username)}
```

Comment:
clicking the checkbox doesn't do anything but clicking the row does -- i suspect  `_toggleSelected` is getting triggered twice

Reply:
only on desktop

Reply:
unfortunate that this happens on desktop, `checkbox.desktop` should probably have a `stopPropagation()` there. I'll just switch this handler off on desktop.